File: bindings/python/examples/secret_manager/ledger_nano.py
from dotenv import load_dotenv

from iota_sdk import LedgerNanoSecretManager, SecretManager

load_dotenv()

# In this example we will get the ledger status and generate an address
# To use the ledger nano simulator clone https://github.com/iotaledger/ledger-shimmer-app, run `git submodule init && git submodule update --recursive`,
# then `./build.sh -m nanos|nanox|nanosplus -s` and use `True` in
# `LedgerNanoSecretManager(True)`.

is_simulator = True

ledger_secret_manager = LedgerNanoSecretManager(is_simulator)
secret_manager = SecretManager(ledger_secret_manager)

# Get the Ledger Nano status.
ledger_nano_status = secret_manager.get_ledger_nano_status()

print(f'Ledger Nano status: {ledger_nano_status}')

# Generate public address with custom account index and range.
address = secret_manager.generate_ed25519_addresses(
    account_index=0, start=0, end=1)

print(f'Address: {address[0]}')

File: bindings/nodejs/examples/secret_manager/ledger-nano.ts
// Copyright 2021-2023 IOTA Stiftung
// SPDX-License-Identifier: Apache-2.0

import { initLogger, SecretManager } from '@iota/sdk';
require('dotenv').config({ path: '.env' });

// Run with command:
// yarn run-example ./secret_manager/ledger-nano.ts

// In this example we will get the ledger status and generate an address
// To use the ledger nano simulator clone https://github.com/iotaledger/ledger-iota-app, run `git submodule init && git submodule update --recursive`,
// then `./build.sh -m nanos|nanox|nanosplus -s` and use `true` in `LedgerSecretManager::new(true)`.
async function run() {
    initLogger();

    try {
        const isSimulator = false;

        const ledgerNanoSecretManager = new SecretManager({
            ledgerNano: isSimulator,
        });

        const ledgerNanoStatus =
            await ledgerNanoSecretManager.getLedgerNanoStatus();

        console.log(ledgerNanoStatus);

        const address = await ledgerNanoSecretManager.generateEd25519Addresses({
            accountIndex: 0,
            range: {
                start: 0,
                end: 1,
            },
        });

        console.log('First public address:', address, '\n');
    } catch (error) {
        console.error('Error: ', error);
    }
}

run().then(() => process.exit());
